### ISSUE TITLE
Prevent contract name collision with @openzeppelin/contracts

### ISSUE THREAD

**Issue by montyly** [OpenZeppelin/openzeppelin-contracts-upgradeable#82]
Hi,

**🧐 Motivation**
`@openzeppelin/contracts-ethereum-package` and `@openzeppelin/contracts` share many contracts names. The duplication prevents the correct compilation artifacts to be generated if a codebase uses both packages.

**📝 Details**
Most of the compilation frameworks do not handle well codebase with contract's name duplicate. For example, when generating the json artifacts, Truffle will not be able to generate correctly the files https://github.com/trufflesuite/truffle/issues/1087. 

As a result, it can prevent the integration with third-party tools. For example, we had to develop a specific [detector in Slither](https://github.com/crytic/slither/wiki/Detector-Documentation#name-reused) to know what contracts names are reused. But if a codebase uses both `@openzeppelin/contracts-ethereum-package` and `@openzeppelin/contracts`, it is difficult for a user to avoid the duplicate names.

**Example**
A codebase that would have both an upgradeable ERC20 token a non-upgradeable one will be affected:
```solidity
// contracts/UpgradeableERC20.sol
import '@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/ERC20.sol';

contract UpgradeableERC20 is ERC20 {
}
```
```solidity
// contracts/NonUpgradeableERC20.sol
import "@openzeppelin/contracts/token/ERC20/ERC20.sol";

contract NonUpgradeableERC20 is ERC20 {
}
```

Running `truffle compile`  on a codebase with these two files will result in one `build/contracts/ERC20.json` file that will either represent `@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/ERC20.sol` or `@openzeppelin/contracts/token/ERC20/ERC20.sol`.

A better practice would be to have separate contract names according to the package. 

Let me know if you have any questions,
Thanks!

**Comment by frangio** [comment]
Thanks for this suggestion! It's a good idea and we're going to consider this for the upcoming 3.0 release.

**Comment by frangio** [comment]
This has been partially fixed in 3.0 but interfaces still have names without suffix. We'll have to correct this in a later release.

**Comment by frangio** [comment]
Fixed in `@openzeppelin/contracts-upgradeable@3.2.0`.

See [First release of OpenZeppelin Contracts Upgradeable](https://forum.openzeppelin.com/t/first-release-of-openzeppelin-contracts-upgrade-safe/4548).
